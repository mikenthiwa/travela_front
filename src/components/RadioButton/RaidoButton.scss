.trip-align {
  display: flex;
}
ul {
  list-style-type: none;
  padding: 0 0 0 10px;
}

ul li {
  color: black;
  position: relative;
  float: left;
}

ul li input[type="radio"] {
  position: absolute;
  visibility: hidden;
}

ul li label {
  position: relative;

  font-size: 1em;
  font-family: "DIN Pro";
  padding: 16px 49px 25px 32px;
  margin: 10px auto;
  height: 30px;
  z-index: 9;
  cursor: pointer;
}

ul li .check {
  display: block;
  position: absolute;
  border: 2px solid #bcbcbc;
  border-radius: 100%;
  height: 20px;
  width: 20px;
  top: 0px;
  left: -1px;
  z-index: 5;
  transition: border 0.25s linear;
}

ul li .check::before {
  display: block;
  position: absolute;
  content: "";
  border-radius: 100%;
  height: 10px;
  width: 10px;
  top: 5px;
  left: 5px;
  margin: auto;
}

input[type="radio"]:checked ~ .check {
  border: 2px solid #3359db;
}

input[type="radio"]:checked ~ .check::before {
  background: #3359db;
}

input[type="radio"]:checked ~ label {
  color: black;
}

@media(max-width: 768px){
  .trip-align {
    display: flex;
    flex-direction: column;
  }
}
