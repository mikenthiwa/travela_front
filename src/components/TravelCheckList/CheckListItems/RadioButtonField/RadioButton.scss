.input-radio-button {
  border-radius: 50%;
  background: white;
  width: 15px;
  height: 15px;
  cursor: pointer;
  position: relative;
  border: 1px solid #ccc;
  box-shadow: 1px 1px 1px 1px rgba(0, 0, 0, 0.1);

  &:before {
    position: absolute;
    border-radius: 50%;
    content: '';
    background: #3359db;
    width: 7px;
    height: 7px;
    z-index: 10;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%) scale(0);
    transition: all 0.2s;
  }
  &.checked {
    border: 1px solid #3359db;

    &:before {
      transform: translate(-50%, -50%) scale(1);
    }
  }
}
