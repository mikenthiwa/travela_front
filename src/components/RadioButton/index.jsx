import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './RaidoButton.scss';

class RadioButton extends Component {
  render() {
    const  { name, value, id, defaultChecked} = this.props;
    return (
      <ul style={{ padding: '0' }}>
        <li>
          <input
            type="radio"
            id={id}
            value={value}
            name="selector"
            checked={defaultChecked}
            onChange={()=>{}}
          />
          <label htmlFor={id}>
            {name}
          </label>
          <div className="check" />
        </li>
      </ul>
    );
  }
}

RadioButton.propTypes = {
  name: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  value:  PropTypes.string,
  id:  PropTypes.string,
  defaultChecked: PropTypes.bool,
};

RadioButton.defaultProps = {
  name: '',
  value:  '',
  id: '',
  defaultChecked: '',
};


export default RadioButton;
