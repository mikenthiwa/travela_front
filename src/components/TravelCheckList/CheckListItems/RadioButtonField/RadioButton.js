import React from 'react';
import PropTypes from 'prop-types';
import './RadioButton.scss';

class RadioButton extends React.Component {
  state = {
    checked: false
  };

  componentWillMount = () => {
    const { checked } = this.props;
    this.setState({ checked });
  };

  componentWillReceiveProps(nextProps) {
    const { checked } = nextProps;
    this.setState({ checked });
  }

  handleClick = () => {
    const { value, onChange } = this.props;
    this.setState(
      prevState => ({ checked: !prevState.checked }),
      () => {
        const { checked } = this.state;
        onChange({
          target: {
            ...this.props,
            checked,
            value: checked ? value : null
          }
        });
      }
    );
  };

  render() {
    const { checked } = this.state;
    return (
      <div
        role="presentation"
        className={`input-radio-button ${checked ? 'checked' : ''}`}
        onClick={this.handleClick}
        onKeyDown={() => {}}
      />
    );
  }
}

RadioButton.propTypes = {
  checked: PropTypes.bool.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default RadioButton;
