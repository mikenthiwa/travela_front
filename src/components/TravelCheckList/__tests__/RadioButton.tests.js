import React from 'react';
import { mount } from 'enzyme';
import sinon from 'sinon';
import RadioButton from '../CheckListItems/RadioButtonField/RadioButton';

describe('<RadioButton />', () => {
  const props = {
    onChange: sinon.spy(),
    checked: false,
    value: ''
  };

  const wrapper = mount(<RadioButton {...props} />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should set the correct state when the component mounts', () => {
    expect(wrapper.instance().state).toEqual({
      checked: false
    });
  });
  it('should render radio button', () => {
    expect(wrapper.find('.input-radio-button').length).toBe(1);
    expect(wrapper.find('.input-radio-button').hasClass('checked')).toEqual(false);
  });

  it('should check the button onClick', () => {
    const radioButton = wrapper.find('.input-radio-button');
    radioButton.simulate('click');
    expect(wrapper.find('.input-radio-button').hasClass('checked')).toEqual(true);
    radioButton.simulate('click');
    expect(wrapper.find('.input-radio-button').hasClass('checked')).toEqual(false);
  });

  it('should check the butto on props change', () => {
    wrapper.setProps({ checked: true})
    expect(wrapper.find('.input-radio-button').hasClass('checked')).toEqual(true);
  });
});
