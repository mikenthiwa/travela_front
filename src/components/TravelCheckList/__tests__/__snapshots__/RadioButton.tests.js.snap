// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> should match snapshot 1`] = `
<div
  className="input-radio-button "
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
/>
`;
